### PR TITLE
dialyze-apps: cleanse Dialyzer warnings for applications/

### PR DESCRIPTION
**Pull request by fenollp**
warn_umatched_return          webhooks_util.erl:392: Expression produces a value of type ['ok'], but this value is unmatched
warn_umatched_return          webhooks_util.erl:490: Expression produces a value of type ['ok'], but this value is unmatched
warn_umatched_return          trunkstore_maintenance.erl:32: Expression produces a value of type [any()], but this value is unmatched
warn_umatched_return          sysconf_acls.erl:182: Expression produces a value of type ['ok'], but this value is unmatched
warn_matching                 stepswitch_resources.erl:288: Clause guard cannot succeed. The pattern <IP, Port, _, Gateway = {'gateway', _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, 'true'}> was matched against the type <_,'undefined' | integer(),_,_>
warn_failing_call             spyvsspy_handlers.erl:102: The call spyvsspy_handlers:respond_error(JObj::any(),'timeout') breaks the contract (wh_json:object(),ne_binary()) -> 'ok'
warn_umatched_return          gcm.erl:85: Expression produces a value of type 'ok' | {'error',_} | {'ok',timer:tref()}, but this value is unmatched
warn_umatched_return          apns.erl:52: Expression produces a value of type 'ok' | {'error',_}, but this value is unmatched
warn_umatched_return          apns_connection.erl:190: Expression produces a value of type ['ok'], but this value is unmatched
warn_umatched_return          pivot_call.erl:107: Expression produces a value of type 'ok' | [any()], but this value is unmatched
warn_failing_call             pivot_call.erl:249: The call pivot_call:maybe_debug_req(Call::any(),Url::'undefined' | binary(),'post',Headers::[{[45 | 67 | 84 | 101 | 110 | 111 | 112 | 116 | 121,...],[1..255,...]},...],Body::any(),Debug::boolean()) breaks the contract (whapps_call:call(),iolist(),atom(),wh_proplist(),iolist(),boolean()) -> 'ok'
warn_failing_call             pivot_call.erl:317: The call pivot_call:maybe_debug_resp(Debug::boolean(),Call::any(),any(),RespHeaders::[{atom() | <<_:8,_:_*8>> | fun() | [<<_:8,_:_*8>> | char()] | number(),_}],RespBody::'undefined' | binary()) breaks the contract (boolean(),whapps_call:call(),ne_binary(),ne_binary(),[any()]) -> 'ok'
warn_failing_call             pivot_call.erl:368: The call pivot_call:maybe_debug_resp(Debug::boolean(),Call::any(),RespCode::'undefined' | binary(),RespHeaders::'undefined' | [binary()],RespBody::'undefined' | binary()) breaks the contract (boolean(),whapps_call:call(),ne_binary(),ne_binary(),[any()]) -> 'ok'
warn_contract_types           pivot_call.erl:569: Invalid type specification for function pivot_call:maybe_debug_resp/5. The success typing is (boolean(),_,_,'undefined' | [binary() | {atom() | <<_:8,_:_*8>> | fun() | [any()] | number(),_}],'undefined' | binary()) -> 'ok'
warn_matching                 konami_code_fsm.erl:90: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:91: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:137: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:144: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:152: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:159: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:194: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:201: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:278: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:455: The pattern 'true' can never match the type 'false'
warn_matching                 konami_code_fsm.erl:462: The pattern 'true' can never match the type 'false'
warn_umatched_return          konami_event_listener.erl:211: Expression produces a value of type [any()], but this value is unmatched
warn_umatched_return          konami_event_listener.erl:212: Expression produces a value of type [any()], but this value is unmatched
warn_umatched_return          konami_event_listener.erl:217: Expression produces a value of type [any()], but this value is unmatched
warn_umatched_return          konami_event_listener.erl:376: Expression produces a value of type [any()], but this value is unmatched
warn_umatched_return          konami_maintenance.erl:143: Expression produces a value of type ['ok'], but this value is unmatched
warn_contract_types           fax_worker.erl:758: Invalid type specification for function fax_worker:get_sizes/1. The success typing is ('undefined') -> {_,non_neg_integer()}
warn_matching                 ecallmgr_fs_notify.erl:241: The pattern 'undefined' can never match the type binary()
warn_matching                 ecallmgr_fs_notify.erl:247: The pattern 'undefined' can never match the type binary()
warn_contract_types           crossbar_maintenance.erl:809: Invalid type specification for function crossbar_maintenance:find_apps/1. The success typing is (atom() | [atom() | [any()] | char()]) -> any()
warn_matching                 crossbar_maintenance.erl:887: The pattern [{IconId, IconData} | _] can never match the type []
warn_not_called               crossbar_maintenance.erl:891: Function add_icon/4 will never be called
warn_failing_call             crossbar_maintenance.erl:908: The call crossbar_maintenance:read_icon(IconPath::binary() | string(),Icon::string()) breaks the contract (ne_binary(),ne_binary()) -> binary()
warn_contract_types           crossbar_maintenance.erl:910: Invalid type specification for function crossbar_maintenance:read_icon/2. The success typing is (binary() | string(),string()) -> binary()

warn_bin_construction         stepswitch_resources.erl:982: Binary construction will fail since the value field Port in segment Port/binary-unit:8 has type 'undefined' | integer()

warn_umatched_return          acdc_agent_sup.erl:50: Expression produces a value of type 'ok' | {'error','not_found'}, but this value is unmatched
warn_umatched_return          cf_voicemail.erl:723: Expression produces a value of type 'false' | {'error',_} | {'ok',_}, but this value is unmatched
warn_umatched_return          cf_voicemail.erl:728: Expression produces a value of type 'false' | {'error',_} | {'ok',_}, but this value is unmatched
warn_umatched_return          cf_voicemail.erl:733: Expression produces a value of type 'false' | {'error',_} | {'ok',_}, but this value is unmatched
warn_umatched_return          cf_park.erl:119: Expression produces a value of type {'error',_} | {'ok',_}, but this value is unmatched
warn_umatched_return          cf_hotdesk.erl:80: Expression produces a value of type 'ok' | {'error',_} | {'ok',_}, but this value is unmatched
warn_umatched_return          cf_hotdesk.erl:84: Expression produces a value of type 'ok' | {'error',_} | {'ok',_}, but this value is unmatched
warn_umatched_return          camper_offnet_handler.erl:198: Expression produces a value of type {'error',_} | {'ok',timer:tref()}, but this value is unmatched
warn_umatched_return          camper_offnet_handler.erl:205: Expression produces a value of type {'error',_} | {'ok','cancel'}, but this value is unmatched
warn_matching                 cdr_v3_migrate_worker.erl:52: The pattern {'error', _E} can never match the type 'ok'
warn_matching                 cdr_v3_migrate_worker.erl:71: The pattern Error = {'error', _} can never match the type 'ok'
warn_failing_call             cdr_v3_migrate_worker.erl:85: The call filelib:ensure_dir(ArchivePath::binary()) breaks the contract (Name) -> 'ok' | {'error',Reason} when is_subtype(Name,filename() | dirname()), is_subtype(Reason,file:posix())
warn_contract_types           cdr_v3_migrate_worker.erl:180: Invalid type specification for function cdr_v3_migrate_worker:create_view_options/1. The success typing is ({non_neg_integer(),1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12,1..255}) -> ['include_docs' | {<<_:48,_:_*16>>,non_neg_integer()},...]
warn_failing_call             csv_util.erl:66: The call filelib:ensure_dir(TestPath::binary()) breaks the contract (Name) -> 'ok' | {'error',Reason} when is_subtype(Name,filename() | dirname()), is_subtype(Reason,file:posix())
warn_contract_types           cdr_sup.erl:38: Invalid type specification for function cdr_sup:get_migrate_server_pid/1. The success typing is (pid()) -> none()
warn_failing_call             cdr_sup.erl:40: The call cdr_sup:child_of_type(Super::'cdr_sup','cdr_v3_migrator') breaks the contract (pid(),atom()) -> pids()
warn_contract_types           cdr_sup.erl:45: Invalid type specification for function cdr_sup:child_of_type/2. The success typing is ('cdr_sup','cdr_v3_migrator') -> ['restarting' | 'undefined' | pid()]
warn_umatched_return          cdr_sup.erl:81: Expression produces a value of type 'ok' | {'error','not_found' | 'simple_one_for_one'}, but this value is unmatched
warn_umatched_return          ecallmgr_call_control.erl:958: Expression produces a value of type queue(), but this value is unmatched
warn_umatched_return          fax_xmpp_sup.erl:48: Expression produces a value of type 'ok' | {'error','not_found' | 'simple_one_for_one'}, but this value is unmatched
warn_umatched_return          whistle_snmp.erl:38: Expression produces a value of type 'ok' | {'error',_}, but this value is unmatched
warn_failing_call             pivot_call.erl:442: The call pivot_call:send(UpdatedCall::any(),maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),'get',[],[],Debug::boolean()) will never return since it differs in the 2nd argument from the success typing arguments: (any(),binary(),'get' | 'post',[{[any(),...],[any(),...]}],maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),boolean())
warn_contract_types           pivot_call.erl:449: Invalid type specification for function pivot_call:send/6. The success typing is (_,binary(),'get' | 'post',[{[any(),...],[any(),...]}],maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),boolean()) -> {'stop',_} | {'ok',_,_}

warn_matching                 crossbar_maintenance.erl:887: The pattern [{IconId, IconData} | _] can never match the type []
warn_not_called               crossbar_maintenance.erl:891: Function add_icon/4 will never be called
warn_failing_call             crossbar_maintenance.erl:908: The call crossbar_maintenance:read_icon(IconPath::binary() | string(),Icon::string()) breaks the contract (ne_binary(),ne_binary()) -> binary()
warn_contract_types           crossbar_maintenance.erl:910: Invalid type specification for function crossbar_maintenance:read_icon/2. The success typing is (binary() | string(),string()) -> binary()

warn_failing_call             csv_util.erl:66: The call filelib:ensure_dir(TestPath::binary()) breaks the contract (Name) -> 'ok' | {'error',Reason} when is_subtype(Name,filename() | dirname()), is_subtype(Reason,file:posix())
warn_failing_call             crossbar_maintenance.erl:887: The call crossbar_maintenance:add_icon(AppId::any(),MasterAccountDb::<<_:8,_:_*8>>,IconId::[any()],IconData::binary()) breaks the contract (ne_binary(),ne_binary(),ne_binary(),binary()) -> 'ok'
warn_contract_types           crossbar_maintenance.erl:890: Invalid type specification for function crossbar_maintenance:add_icon/4. The success typing is (_,<<_:8,_:_*8>>,[any()],binary()) -> 'ok'
warn_return_only_exit         cb_sup.erl:94: Function system_terminate/4 only terminates with explicit exception
warn_failing_call             pivot_call.erl:443: The call pivot_call:send(UpdatedCall::any(),string(),'get',[],[],Debug::boolean()) will never return since it differs in the 2nd argument from the success typing arguments: (any(),binary(),'get' | 'post',[{[any(),...],[any(),...]}],maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),boolean())
warn_contract_types           pivot_call.erl:450: Invalid type specification for function pivot_call:send/6. The success typing is (_,binary(),'get' | 'post',[{[any(),...],[any(),...]}],maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),boolean()) -> {'stop',_} | {'ok',_,_}
warn_contract_types           fax_worker.erl:758: Invalid type specification for function fax_worker:get_sizes/1. The success typing is ('undefined') -> {_,non_neg_integer()}

warn_failing_call             pivot_call.erl:443: The call pivot_call:send(UpdatedCall::any(),string(),'get',[],[],Debug::boolean()) will never return since it differs in the 2nd argument from the success typing arguments: (any(),binary(),'get' | 'post',[{[any(),...],[any(),...]}],maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),boolean())
warn_contract_types           pivot_call.erl:451: Invalid type specification for function pivot_call:send/6. The success typing is (_,binary(),'get' | 'post',[{[any(),...],[any(),...]}],maybe_improper_list(binary() | maybe_improper_list(any(),binary() | []) | byte(),binary() | []),boolean()) -> {'stop',_} | {'ok',_,_}

warn_failing_call             fax_worker.erl:764: The call filelib:file_size(OutputFile::<<_:8,_:_*8>>) breaks the contract (Filename) -> non_neg_integer() when is_subtype(Filename,filename())
warn_failing_call             csv_util.erl:66: The call filelib:ensure_dir(TestPath::binary()) breaks the contract (Name) -> 'ok' | {'error',Reason} when is_subtype(Name,filename() | dirname()), is_subtype(Reason,file:posix())
warn_opaque                   ecallmgr_fs_channels.erl:643: The call ecallmgr_fs_channels:query_channels('$end_of_table' | {[[any()]],'$end_of_table' | {atom() | tid(),integer(),integer(),binary(),[any()],integer()} | {atom() | tid(),_,_,integer(),binary(),[any()],integer(),integer()}},Fields::any(),any()) contains an opaque term as 1st argument when terms of different types are expected in these positions
warn_opaque                   ecallmgr_fs_channels.erl:648: The call ecallmgr_fs_channels:query_channels('$end_of_table' | {[[any()]],'$end_of_table' | {atom() | tid(),integer(),integer(),binary(),[any()],integer()} | {atom() | tid(),_,_,integer(),binary(),[any()],integer(),integer()}},Fields::any(),any()) contains an opaque term as 1st argument when terms of different types are expected in these positions
warn_contract_types           ecallmgr_fs_channels.erl:653: Invalid type specification for function ecallmgr_fs_channels:query_channels/3. The success typing is ('$end_of_table',_,_) -> any()
warn_matching                 ecallmgr_fs_channels.erl:656: The pattern <{[Channel = {'channel', CallId, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _}], Continuation}, <<97:8/integer-unit:1,108:8/integer-unit:1,108:8/integer-unit:1>>, Channels> can never match the type <{[[any()]],'$end_of_table' | {atom() | tid(),integer(),integer(),binary(),[any()],integer()} | {atom() | tid(),_,_,integer(),binary(),[any()],integer(),integer()}},_,_>
warn_matching                 ecallmgr_fs_channels.erl:663: The pattern <{[Channel = {'channel', CallId, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _}], Continuation}, Fields, Channels> can never match the type <{[[any()]],'$end_of_table' | {atom() | tid(),integer(),integer(),binary(),[any()],integer()} | {atom() | tid(),_,_,integer(),binary(),[any()],integer(),integer()}},_,_>
warn_umatched_return          ecallmgr_fs_conference.erl:172: Expression produces a value of type 'continue' | 'stop' | {'continue',[{<<_:64,_:_*8>>,_},...]}, but this value is unmatched

### CODE REVIEW

File: applications/cdr/src/csv_util.erl [hunk 1]
```diff
@@ -46,7 +46,7 @@ test_convert(AccountDb) ->
                                           [CdrDoc | Acc]
                                   end, [], JObjs),
             CsvData = json_objs_to_csv(CdrDocs),
-            maybe_save_csv(<<"test.csv">>, CsvData)
+            maybe_save_csv("test.csv", CsvData)
```
Comment (jamesaimonetti) [inline applications/cdr/src/csv_util.erl]: `file:filename()` and `file:name()` both accept binaries; why the change?

Comment (fenollp) [inline applications/cdr/src/csv_util.erl]: This was the warning:
```
warn_failing_call csv_util.erl:66: The call filelib:ensure_dir(TestPath::binary()) breaks the contract (Name) -> 'ok' | {'error',Reason} when is_subtype(Name,filename() | dirname()), is_subtype(Reason,file:posix())
```

I have no idea why `binary()` is not accepted by Dialyzer here… I am still trying to have 17's Dialyzer to run against our code base (it gives much less false positives)

File: applications/ecallmgr/src/ecallmgr_fs_notify.erl [hunk 2]
```diff
@@ -215,8 +215,8 @@ register_overwrite(JObj, Props) ->
                    ,Realm
                   ),
     SipUri = nksip_unparse:uri(#uri{user=Username, domain=Realm}),
-    PrevBody = wh_util:to_list(<<"Replaced-By:", NewContact/binary>>),
-    NewBody = wh_util:to_list(<<"Overwrote:", PrevContact/binary>>),
+    PrevBody = wh_util:to_list(<<"Replaced-By:", (wh_util:to_binary(NewContact))/binary>>),
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: Would need to check with @lazedo here; If PrevBody or NewBody are `'undefined'`, should this crash when building this binary like before?

Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: @jamesaimonetti read a bit farther: if `PrevContact` is undefined `Prev` "data" is not sent (same with `NewContact`).

Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: That's what I'm saying though; if `PrevContact` is 'undefined', building `NewBody` would crash, no? I think I just typed it poorly previously.

Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_fs_notify.erl]: `wh_util:to_binary/1` does not fail on atoms (nor on `'undefined'`)
